.github/workflows/test.yml

```diff
@@ -25,10 +25,10 @@ jobs:
         # doesn't touch mongo against multiple mongo versions
         include:
           - java: '8'
-            mongo: 'mongodb-linux-x86_64-3.6.23'
+            mongo: 'mongodb-linux-x86_64-ubuntu2204-7.0.4'
```

Comment:
Only change 1 to v7, we still need to test against v3

Reply:
Okay, that means we need code to be able to work on both versions. Then simply deleting `ns=` would work in v7 but not v3. So, still need to pass in mongoDB version to make failed tests pass.

Reply:
You delete ns from the test expectation as well as the data structure returned from mongo

Reply:
https://www.mongodb.com/docs/manual/release-notes/4.4-compatibility/#general-changes

> Below 4.4 general changes:

>The [listIndexes](https://www.mongodb.com/docs/manual/reference/command/listIndexes/#mongodb->dbcommand-dbcmd.listIndexes) and the mongo shell helper method [db.collection.getIndexes()](https://www.mongodb.com/docs/manual/reference/method/db.collection.getIndexes/#mongodb-method-db.collection.getIndexes) no longer returns the namespace ns field in the index specification documents.



If my understanding is correct, if dbVersion < 4.4.0, then need to rm `ns`?

Reply:
I would just remove it period. It's not an important part of the test; I just included it because it was there. The index specs are the thing we care about

Reply:
so you are saying we just remove it if it exists regardless dbversion?

Reply:
yep

Reply:
[junit] Testcase: indexesTokens took 0.328 sec
    [junit]     FAILED
    [junit] incorrect indexes
    [junit] Expected: is <[Document{{v=2, key=Document{{user=1}}, name=user_1}}, Document{{v=2, key=Document{{_id=1}}, name=_id_}}, Document{{v=2, unique=true, key=Document{{id=1}}, name=id_1}}, Document{{v=2, unique=true, key=Document{{token=1}}, name=token_1}}, Document{{v=2, key=Document{{expires=1}}, name=expires_1, expireAfterSeconds=0}}]>
    [junit]      but: was <[Document{{v=2, key=Document{{user=1}}, name=user_1}}, Document{{v=2, key=Document{{_id=1}}, name=_id_}}, Document{{v=2, key=Document{{id=1}}, name=id_1, unique=true}}, Document{{v=2, key=Document{{token=1}}, name=token_1, unique=true}}, Document{{v=2, key=Document{{expires=1}}, name=expires_1, expireAfterSeconds=0}}]>

It seems the order of the keys in each document also matter. This is kinda strange ...

Reply:
I think Document preserves insertion order IIRC...?

Reply:
Hm, maybe Document preserves insert order? Not at all sure about that
Or maybe the integer types don't match. int(1) != long(1)

Reply:
`expireAfterSeconds` is causing the problem. long(0) in v3 vs int(0) in v7.
If this field is not important, I would like to remove it as well.

Reply:
That one is important as we're using TTL indexes for the tokens collection. I'd just go through the mongo response and convert everything to longs

Reply:
I don't know what else I can try. I convert everything(numeric) to longs. V7 passed but v3 did not.

Reply:
Put in some temporary code to run through each key / value and see which ones aren't equal. You could also check if Document is insert order sensitive

Reply:
I put in some temp code in [indexesConfigApp](https://github.com/kbase/auth2/blob/dev-mongodb_upgrade/src/us/kbase/test/auth2/lib/storage/mongo/MongoStorageStartUpTest.java#L170) and printed out all key, value, and value type.
It's still kinda odd to me this test failed in v3. 

<img width="952" alt="Screen Shot 2024-01-16 at 2 20 43 PM" src="https://github.com/kbase/auth2/assets/22202096/991c4e71-d8c9-4c2a-94c9-da231cb86238">


GHA: https://github.com/kbase/auth2/actions/runs/7515050478/job/20458587218?pr=415#step:6:115

Reply:
Updated!

Reply:
:+1: